modular_citadel/code/modules/reagents/reagents/cit_reagents.dm

```diff
@@ -118,21 +122,26 @@
 	overdose_threshold = 20
 
 /datum/reagent/drug/aphrodisiacplus/on_mob_life(mob/living/M)
-	if(M && M.canbearoused && !(M.client?.prefs.cit_toggles & NO_APHRO))
-		if(prob(33))
-			M.adjustArousalLoss(6)//not quite six times as powerful, but still considerably more powerful.
+	if(M && M.client?.prefs.arousable && !(M.client?.prefs.cit_toggles & NO_APHRO))
 		if(prob(5))
-			if(M.getArousalLoss() > 75)
+			if(prob(current_cycle))
 				M.say(pick("Hnnnnngghh...", "Ohh...", "Mmnnn..."))
 			else
 				M.emote(pick("moan","blush"))
 		if(prob(5))
 			var/aroused_message
-			if(M.getArousalLoss() > 90)
+			if(current_cycle>25)
 				aroused_message = pick("You need to fuck someone!", "You're bursting with sexual tension!", "You can't get sex off your mind!")
 			else
 				aroused_message = pick("You feel a bit hot.", "You feel strong sexual urges.", "You feel in the mood.", "You're ready to go down on someone.")
 			to_chat(M, "<span class='userlove'>[aroused_message]</span>")
+			REMOVE_TRAIT(M,TRAIT_NEVERBONER,"aphro")
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			var/list/genits = H.adjust_arousal(100, aphro = TRUE) // redundant but should still be here
+			for(var/g in genits)
+				var/obj/item/organ/genital/G = g
+				to_chat(M, "<span class='userlove'>[G.arousal_verb]!</span>")
```

Comment:
Spammy.

Reply:
It should be noted that it just arousal only returns the genitals that it actually adjusted

Reply:
Umh, ok.